### ISSUE TITLE
Pipeline "Task not serializable" on EMR and Databricks

### ISSUE THREAD

**Issue by roblovelock** [JohnSnowLabs/spark-nlp#679]
## Description
I am trying to run a simple pipeline on Databricks Runtime 6.1 ML (I have also tried this on EMR emr-5.28.0 Zeppelin). There is an issue running the pipeline when data is loaded via `spark.read`.

```
import com.johnsnowlabs.nlp.annotator._
import com.johnsnowlabs.nlp.base._
import com.johnsnowlabs.util.{Benchmark}
import org.apache.spark.ml.Pipeline

val dataIn = Seq(
    "Mixed green salad and tea.  I have IBS.  Can't choose something nutty if it's an entire month."
).toDS.toDF("message")

//Save the data for use later
dataIn.write.mode("overwrite").parquet("/tmp/sparknlp/testpipeline")

val documentAssembler = new DocumentAssembler()
    .setInputCol("message")
    .setOutputCol("document")
 
val sentence_detector = new SentenceDetector()
    .setInputCols(Array("document"))
    .setOutputCol("sentence")

val tokenizer = new Tokenizer()
    .setInputCols(Array("sentence"))
    .setOutputCol("token")

val pipeline = new Pipeline()
    .setStages(Array(
        documentAssembler,
        sentence_detector,
        tokenizer
    ))

val model = Benchmark.time("Time to train Model") {
  pipeline.fit(dataIn)
}
val dataOut = model.transform(dataIn)

//Working
dataOut.show()

val dataIn2 = spark.read.parquet("/tmp/sparknlp/testpipeline")

val model2 = Benchmark.time("Time to train Model") {
  pipeline.fit(dataIn2)
}
val dataOut2 = model2.transform(dataIn2)

//Not Working
dataOut2.show()
```

## Expected Behavior
I expect the result from `dataOut2.show()` to be the same as `dataOut.show()` as the input dataframe is the same.

## Current Behavior

`org.apache.spark.SparkException: Task not serializable` ... `Caused by: java.io.UTFDataFormatException: encoded string too long: 66950 bytes`

## Possible Solution
Maybe something to do with Typesafe confg??? 
https://stackoverflow.com/questions/41505599/task-not-serializable-in-spark-caused-by-utfdataformatexception-encoded-string

## Steps to Reproduce
1. Start Databricks 6.1 ML cluster
2. Install Spark-nlp version 2.3.2 (issue happens on prev 2.3.x versions as well)
3. run the code above

## Context
I can't use Spark-nlp on Databricks or EMR. Originally I was trying to use Bert Embeddings, but can't even get a simple pipeline working.

## Your Environment
* Version used: 2.3.2
* Browser Name and version: Chrome
* Operating System and version (desktop or mobile): desktop Mint Linux
* Databricks Runtime: 6.1 ML (includes Apache Spark 2.4.4, Scala 2.11)
* EMR: emr-5.28.0

**Comment by maziyarpanahi** [comment]
I can think of two issues, first is ML 6.1 that was reported before and second is if you are starting your cluster as YARN cluster-mode and not client-mode.
Could you please try client-mode?

**Comment by roblovelock** [comment]
Thanks for your feedback. I can't find any issues mentioning ML 6.1. Can you point me in the direction of where this issue was raised and what the work around was?

I am able to run the simple pipeline on ML 5.5, however there is still an issue running the Bert Embeddings. There is an issue trying to load your `*.so` files into tensorflow:

`java.lang.UnsatisfiedLinkError: /local_disk0/spark-7e30ecb9-3fa3-40b8-aa35-a42464618da3/userFiles-c947a5ee-b452-41aa-b8ff-ed0e44a92fb8/sparknlp_contrib_spar: undefined symbol: _ZN10tensorflow14kernel_factory17OpKernelRegistrar12InitInternalEPKNS_9KernelDefEN4absl11string_viewEPFPNS_8OpKernelEPNS_20OpKernelConstructionEE`

It may be intentional, but when you load libraries into tensorflow you set `com.johnsnowlabs.nlp.annotators.ner.dl.LoadsContrib.loadedToTensorflow = false` at the start of the method. This means that if it fails it won't attempt to load the files again, but as it failed first time they won't be available for use.

My work around for Bert Embeddings on Databricks is to: 
* run a 5.5 ML cluster
* run this command `com.johnsnowlabs.nlp.annotators.ner.dl.LoadsContrib.loadContribToTensorflow()`, which will fail
* Now I can use the embeddings as intended.

I'm not creating a ner model so this works for me, I don't what the libraries loaded are needed for, but I suspect an ner model will fail (this is an assumption based on package name).

**Comment by maziyarpanahi** [comment]
That is a different error which you should open a new issue for it and provide the code and a way to reproduce it. 
The ML 6.1 is not compatible with our library for some reason which I am guessing it has something to do with being run in cluster-mode instead of client-mode. (one or two users in our Slack mentioned this and they solved it by using ML 5.x)

**Comment by victorconan** [comment]
> That is a different error which you should open a new issue for it and provide the code and a way to reproduce it.
> The ML 6.1 is not compatible with our library for some reason which I am guessing it has something to do with being run in cluster-mode instead of client-mode. (one or two users in our Slack mentioned this and they solved it by using ML 5.x)

confirmed runtime ML 5.X works while runtime 6.X doesnt.

**Comment by maziyarpanahi** [comment]
Thanks for the confirmation. I have tested the latest releases on 6.2 and 6.3. It works without problem as long as the YARN is in client-mode. However, it fails in ML editions for unknown reasons. (I couldn’t find any difference between ML and none ML Databricks)

https://github.com/JohnSnowLabs/spark-nlp-workshop/tree/master/databricks#compatibility

**Comment by maziyarpanahi** [comment]
This has been resolved in 2.4.5 release: https://github.com/JohnSnowLabs/spark-nlp/releases/tag/2.4.5
